.travis.yml

```diff
@@ -1,14 +1,49 @@
 language: rust
-services: docker
-sudo: required
+cache: cargo
 
-matrix:
-  include:
-    - rust: 1.27.0
-    - rust: stable
-    - rust: nightly
+rust:
+  - 1.27.0
+  - stable
+  - nightly
+
+install:
+  - rustup component add rustfmt
+  - rustup component add clippy
+  - rustup target add thumbv7em-none-eabihf
+  - rustup target add wasm32-unknown-unknown
```

Comment:
Wouldn't installing everything for each run make total CI job duration longer? Plus it causes failure on Rust 1.27.

Reply:
Fixed in d8af372